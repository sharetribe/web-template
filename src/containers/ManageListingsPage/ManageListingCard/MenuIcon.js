import React from 'react';
import classNames from 'classnames';

import css from './ManageListingCard.module.css';

/**
 * Menu icon
 *
 * @param {Object} props
 * @param {string} [props.className] - Custom class that extends the default class for the root element
 * @param {boolean} [props.isActive] - Whether the menu is active
 * @returns {JSX.Element} Menu icon components
 */
const MenuIcon = props => {
  const { className, isActive = false } = props;
  const classes = classNames(css.menuIcon, className);
  const filter = isActive ? '' : 'url(#a)';
  return (
    <svg
      className={classes}
      width="26"
      height="12"
      viewBox="0 0 26 12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <filter x="-38.9%" y="-125%" width="177.8%" height="450%" filterUnits="objectBoundingBox">
          <feOffset dy="2" in="SourceAlpha" result="shadowOffsetOuter1" />
          <feGaussianBlur stdDeviation="2" in="shadowOffsetOuter1" result="shadowBlurOuter1" />
          <feColorMatrix values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.5 0" in="shadowBlurOuter1" />
        </filter>
      </defs>
      <g transform="translate(-342 -18)" filter={filter}>
        <path d="M348 24c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2z" />
      </g>
      <g transform="translate(-342 -18)">
        <path d="M348 24c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2z" />
      </g>
    </svg>
  );
};

export default MenuIcon;
