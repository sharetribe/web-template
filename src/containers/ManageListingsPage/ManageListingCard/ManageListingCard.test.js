import React from 'react';
import '@testing-library/jest-dom';

import { types as sdkTypes } from '../../../util/sdkLoader';
import {
  getHostedConfiguration,
  renderWithProviders as render,
  testingLibrary,
} from '../../../util/testHelpers';
import { createOwnListing, createStock, fakeIntl } from '../../../util/testData';

import { ManageListingCard } from './ManageListingCard';

const { screen, waitFor } = testingLibrary;

const { Money } = sdkTypes;
const noop = () => null;

const getConfig = listingTypes => {
  const hostedConfig = getHostedConfiguration();
  return {
    ...hostedConfig,
    listingTypes: {
      listingTypes,
    },
  };
};

describe('ManageListingCard', () => {
  it('Inquiry (inquiry): no price', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      availabilityPlan: null,

      publicData: {
        listingType: 'free-inquiry',
        transactionProcessAlias: 'default-inquiry/release-1',
        unitType: 'inquiry',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });

    const config = getConfig([
      {
        id: 'free-inquiry',
        transactionProcess: {
          name: 'default-inquiry',
          alias: 'default-inquiry/release-1',
        },
        unitType: 'inquiry',
        defaultListingFields: {
          price: false,
        },
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.queryByText('ManageListingCard.priceNotSet')).not.toBeInTheDocument();
    expect(tree.queryByText('10')).not.toBeInTheDocument(); //fakeIntl
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.queryByText('ManageListingCard.manageAvailability')).not.toBeInTheDocument();
    expect(tree.queryByText('ManageListingCard.manageStock')).not.toBeInTheDocument();
  });

  it('Inquiry (inquiry): with price', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      availabilityPlan: null,

      publicData: {
        listingType: 'free-inquiry',
        transactionProcessAlias: 'default-inquiry/release-1',
        unitType: 'inquiry',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });

    const config = getConfig([
      {
        id: 'free-inquiry',
        transactionProcess: {
          name: 'default-inquiry',
          alias: 'default-inquiry/release-1',
        },
        unitType: 'inquiry',
        defaultListingFields: {
          price: true,
        },
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.queryByText('ManageListingCard.priceNotSet')).not.toBeInTheDocument();
    expect(tree.queryByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.queryByText('ManageListingCard.manageAvailability')).not.toBeInTheDocument();
    expect(tree.queryByText('ManageListingCard.manageStock')).not.toBeInTheDocument();
  });

  it('Booking (day): normal', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      availabilityPlan: {
        type: 'availability-plan/time',
        timezone: 'Etc/UTC',
        entries: [
          { dayOfWeek: 'mon', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'tue', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'wed', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'thu', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'fri', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'sat', startTime: '00:00', endTime: '00:00', seats: 1 },
          //{ dayOfWeek: 'sun', startTime: '00:00', endTime: '00:00', seats: 1 },
        ],
      },

      publicData: {
        listingType: 'rent-bicycles-daily',
        transactionProcessAlias: 'default-booking/release-1',
        unitType: 'hour',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />
    );
    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText('ManageListingCard.perUnit')).toBeInTheDocument();
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageAvailability')).toBeInTheDocument();
  });

  it('Booking (day): closed', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      state: 'closed',
      availabilityPlan: {
        type: 'availability-plan/time',
        timezone: 'Etc/UTC',
        entries: [
          { dayOfWeek: 'mon', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'tue', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'wed', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'thu', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'fri', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'sat', startTime: '00:00', endTime: '00:00', seats: 1 },
          //{ dayOfWeek: 'sun', startTime: '00:00', endTime: '00:00', seats: 1 },
        ],
      },

      publicData: {
        listingType: 'rent-bicycles-daily',
        transactionProcessAlias: 'default-booking/release-1',
        unitType: 'hour',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />
    );
    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText('ManageListingCard.perUnit')).toBeInTheDocument();
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageAvailability')).toBeInTheDocument();

    expect(tree.getByText('ManageListingCard.closedListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.openListing')).toBeInTheDocument();
  });

  it('Booking (day): pendingApproval', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      state: 'pendingApproval',
      availabilityPlan: {
        type: 'availability-plan/time',
        timezone: 'Etc/UTC',
        entries: [
          { dayOfWeek: 'mon', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'tue', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'wed', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'thu', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'fri', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'sat', startTime: '00:00', endTime: '00:00', seats: 1 },
          //{ dayOfWeek: 'sun', startTime: '00:00', endTime: '00:00', seats: 1 },
        ],
      },

      publicData: {
        listingType: 'rent-bicycles-daily',
        transactionProcessAlias: 'default-booking/release-1',
        unitType: 'hour',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />
    );
    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText('ManageListingCard.perUnit')).toBeInTheDocument();
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageAvailability')).toBeInTheDocument();

    expect(tree.getByText('ManageListingCard.pendingApproval')).toBeInTheDocument();
  });

  it('Booking (day): draft', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      state: 'draft',
      availabilityPlan: {
        type: 'availability-plan/time',
        timezone: 'Etc/UTC',
        entries: [
          { dayOfWeek: 'mon', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'tue', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'wed', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'thu', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'fri', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'sat', startTime: '00:00', endTime: '00:00', seats: 1 },
          //{ dayOfWeek: 'sun', startTime: '00:00', endTime: '00:00', seats: 1 },
        ],
      },

      publicData: {
        listingType: 'rent-bicycles-daily',
        transactionProcessAlias: 'default-booking/release-1',
        unitType: 'hour',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText('ManageListingCard.perUnit')).toBeInTheDocument();
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageAvailability')).toBeInTheDocument();

    expect(tree.getByText('ManageListingCard.draftOverlayText')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.finishListingDraft')).toBeInTheDocument();
  });

  it('Booking (day): no price', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: null,
      availabilityPlan: {
        type: 'availability-plan/time',
        timezone: 'Etc/UTC',
        entries: [
          { dayOfWeek: 'mon', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'tue', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'wed', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'thu', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'fri', startTime: '00:00', endTime: '00:00', seats: 1 },
          { dayOfWeek: 'sat', startTime: '00:00', endTime: '00:00', seats: 1 },
          //{ dayOfWeek: 'sun', startTime: '00:00', endTime: '00:00', seats: 1 },
        ],
      },

      publicData: {
        listingType: 'rent-bicycles-daily',
        transactionProcessAlias: 'default-booking/release-1',
        unitType: 'hour',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />
    );
    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.priceNotSet')).toBeInTheDocument();
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageAvailability')).toBeInTheDocument();
  });

  it('Purchase (item): normal', () => {
    const listing = createOwnListing(
      'listing-day',
      {
        title: 'the listing',
        description: 'Lorem ipsum',
        price: new Money(1000, 'USD'),
        availabilityPlan: null,

        publicData: {
          listingType: 'sell-bicycles-item',
          transactionProcessAlias: 'default-purchase/release-1',
          unitType: 'item',
          amenities: ['dog_1'],
          location: {
            address: 'Main Street 123',
            building: 'A 1',
          },
        },
      },
      {
        currentStock: createStock('stock-id', { quantity: 5 }),
      }
    );
    const config = getConfig([
      {
        id: 'sell-bicycles-item',
        label: 'Sell products',
        transactionProcess: {
          alias: 'default-purchase/release-1',
          name: 'default-purchase',
        },
        unitType: 'item',
        stockType: 'multipleItems',
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageStock')).toBeInTheDocument();
  });

  it('Purchase (item): normal out-of-stock', () => {
    const listing = createOwnListing(
      'listing-day',
      {
        title: 'the listing',
        description: 'Lorem ipsum',
        price: new Money(1000, 'USD'),
        availabilityPlan: null,

        publicData: {
          listingType: 'sell-bicycles-item',
          transactionProcessAlias: 'default-purchase/release-1',
          unitType: 'item',
          amenities: ['dog_1'],
          location: {
            address: 'Main Street 123',
            building: 'A 1',
          },
        },
      },
      {
        currentStock: createStock('stock-id', { quantity: 0 }),
      }
    );

    const config = getConfig([
      {
        id: 'sell-bicycles-item',
        label: 'Sell products',
        transactionProcess: {
          alias: 'default-purchase/release-1',
          name: 'default-purchase',
        },
        unitType: 'item',
        stockType: 'multipleItems',
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageStock')).toBeInTheDocument();

    // Out of stock
    expect(tree.getByText('ManageListingCard.outOfStockOverlayText')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.setPriceAndStock')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListingTextOr')).toBeInTheDocument();
  });

  it('Purchase (item): normal', () => {
    const listing = createOwnListing(
      'listing-day',
      {
        title: 'the listing',
        description: 'Lorem ipsum',
        price: new Money(1000, 'USD'),
        availabilityPlan: null,
        state: 'draft',
        publicData: {
          listingType: 'sell-bicycles-item',
          transactionProcessAlias: 'default-purchase/release-1',
          unitType: 'item',
          amenities: ['dog_1'],
          location: {
            address: 'Main Street 123',
            building: 'A 1',
          },
        },
      },
      {
        currentStock: createStock('stock-id', { quantity: 5 }),
      }
    );

    const config = getConfig([
      {
        id: 'sell-bicycles-item',
        label: 'Sell products',
        transactionProcess: {
          alias: 'default-purchase/release-1',
          name: 'default-purchase',
        },
        unitType: 'item',
        stockType: 'multipleItems',
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageStock')).toBeInTheDocument();

    expect(tree.getByText('ManageListingCard.draftOverlayText')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.finishListingDraft')).toBeInTheDocument();
  });

  it('Purchase (item): no price', () => {
    const listing = createOwnListing(
      'listing-day',
      {
        title: 'the listing',
        description: 'Lorem ipsum',
        price: null,
        availabilityPlan: null,

        publicData: {
          listingType: 'sell-bicycles-item',
          transactionProcessAlias: 'default-purchase/release-1',
          unitType: 'item',
          amenities: ['dog_1'],
          location: {
            address: 'Main Street 123',
            building: 'A 1',
          },
        },
      },
      {
        currentStock: createStock('stock-id', { quantity: 5 }),
      }
    );
    const config = getConfig([
      {
        id: 'sell-bicycles-item',
        label: 'Sell products',
        transactionProcess: {
          alias: 'default-purchase/release-1',
          name: 'default-purchase',
        },
        unitType: 'item',
        stockType: 'multipleItems',
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.priceNotSet')).toBeInTheDocument();
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.manageStock')).toBeInTheDocument();
  });

  it('Purchase (item): no stock', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      availabilityPlan: null,

      publicData: {
        listingType: 'sell-bicycles-item',
        transactionProcessAlias: 'default-purchase/release-1',
        unitType: 'item',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });
    const config = getConfig([
      {
        id: 'sell-bicycles-item',
        label: 'Sell products',
        transactionProcess: {
          alias: 'default-purchase/release-1',
          name: 'default-purchase',
        },
        unitType: 'item',
        stockType: 'multipleItems',
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.setPriceAndStock')).toBeInTheDocument();
  });

  it('Purchase (item): oneItem', () => {
    const listing = createOwnListing('listing-day', {
      title: 'the listing',
      description: 'Lorem ipsum',
      price: new Money(1000, 'USD'),
      availabilityPlan: null,

      publicData: {
        listingType: 'sell-bicycles-item',
        transactionProcessAlias: 'default-purchase/release-1',
        unitType: 'item',
        amenities: ['dog_1'],
        location: {
          address: 'Main Street 123',
          building: 'A 1',
        },
      },
    });
    const config = getConfig([
      {
        id: 'sell-bicycles-item',
        label: 'Sell products',
        transactionProcess: {
          alias: 'default-purchase/release-1',
          name: 'default-purchase',
        },
        unitType: 'item',
        stockType: 'oneItem',
      },
    ]);

    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={listing}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );

    expect(tree.getByText('ResponsiveImage.noImage')).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.closeListing')).toBeInTheDocument();
    expect(tree.getByText('10')).toBeInTheDocument(); //fakeIntl
    expect(tree.getByText(/the Listing/i)).toBeInTheDocument();
    expect(tree.getByText('ManageListingCard.editListing')).toBeInTheDocument();
    expect(tree.queryByText('ManageListingCard.setPriceAndStock')).not.toBeInTheDocument();
  });

  it('matches snapshot (purchase)', () => {
    const config = getConfig([
      {
        id: 'sell-bicycles-item',
        label: 'Sell products',
        transactionProcess: {
          alias: 'default-purchase/release-1',
          name: 'default-purchase',
        },
        unitType: 'item',
        stockType: 'multipleItems',
      },
    ]);
    const tree = render(
      <ManageListingCard
        history={{ push: noop }}
        listing={createOwnListing('listing1', {
          publicData: {
            listingType: 'product-selling',
            transactionProcessAlias: 'default-purchase/release-1',
          },
        })}
        intl={fakeIntl}
        isMenuOpen={false}
        onCloseListing={noop}
        onOpenListing={noop}
        onToggleMenu={noop}
        hasClosingError={false}
        hasOpeningError={false}
        availabilityEnabled={true}
      />,
      { config }
    );
    expect(tree.asFragment().firstChild).toMatchSnapshot();
  });
});
