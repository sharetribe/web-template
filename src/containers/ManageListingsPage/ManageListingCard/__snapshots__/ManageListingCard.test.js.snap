// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageListingCard matches snapshot (purchase) 1`] = `
<div
  class="root"
>
  <div
    class="clickWrapper"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        class="aspectPadding"
        style="padding-bottom: 100%;"
      >
        <div
          class="aspectBox"
        >
          <div
            class="rootForImage noImageContainer"
          >
            <div
              class="noImageWrapper"
            >
              <svg
                class="noImageIcon"
                height="49"
                viewBox="0 -1 48 49"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M13.6 9.6L24 .8l10.4 8.8z"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M.8 46.4h46.4V9.6H.8z"
                    stroke-width="2"
                  />
                  <path
                    d="M5.6 41.6h36.8V14.4H5.6z"
                    stroke-width="2"
                  />
                  <path
                    d="M19.998 31.636l.835-.835a4.035 4.035 0 0 1 2.853-1.183h.693c1.07 0 2.097.426 2.853 1.183l.835.835"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M21.599 23.983a1.009 1.009 0 1 0 0 2.018 1.009 1.009 0 0 0 0-2.018M26.402 23.983a1.009 1.009 0 1 0 0 2.018 1.009 1.009 0 0 0 0-2.018"
                  />
                </g>
              </svg>
              <div
                class="noImageText"
              >
                <span>
                  ResponsiveImage.noImage
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="menuOverlayWrapper"
    >
      <div
        class="menuOverlay"
      />
    </div>
    <div
      class="menubarWrapper"
    >
      <div
        class="menubarGradient"
      />
      <div
        class="menubar"
      >
        <div
          class="root menu cardIsOpen"
        >
          <button
            class="root menuLabel"
          >
            <div
              class="iconWrapper"
            >
              <svg
                class="menuIcon menuIcon"
                height="12"
                viewBox="0 0 26 12"
                width="26"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <filter
                    filterUnits="objectBoundingBox"
                    height="450%"
                    width="177.8%"
                    x="-38.9%"
                    y="-125%"
                  >
                    <feoffset
                      dy="2"
                      in="SourceAlpha"
                      result="shadowOffsetOuter1"
                    />
                    <fegaussianblur
                      in="shadowOffsetOuter1"
                      result="shadowBlurOuter1"
                      stdDeviation="2"
                    />
                    <fecolormatrix
                      in="shadowBlurOuter1"
                      values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.5 0"
                    />
                  </filter>
                </defs>
                <g
                  filter="url(#a)"
                  transform="translate(-342 -18)"
                >
                  <path
                    d="M348 24c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2z"
                  />
                </g>
                <g
                  transform="translate(-342 -18)"
                >
                  <path
                    d="M348 24c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2zm7 0c-1.105 0-2-.895-2-2s.895-2 2-2 2 .895 2 2-.895 2-2 2z"
                  />
                </g>
              </svg>
            </div>
          </button>
          <div
            class="menuContent isClosed"
            style="left: -10000px;"
          >
            <ul
              class="content"
            >
              <li
                class="root"
                role="menuitem"
              >
                <button
                  class="menuItem inlineTextButton"
                >
                  <span>
                    ManageListingCard.closeListing
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="info"
  >
    <div
      class="price"
    >
      <div
        class="priceValue"
        title="55"
      >
        55
      </div>
    </div>
    <div
      class="mainInfo"
    >
      <div
        class="titleWrapper"
      >
        <button
          class="title inlineTextButton"
        >
          <span
            style="word-break: break-all;"
          >
            listing1
          </span>
           title
        </button>
      </div>
    </div>
    <div
      class="manageLinks"
    >
      <a
        class="manageLink"
        href="/listing/listing1-title/listing1/edit/upload"
      >
        <span>
          ManageListingCard.editListing
        </span>
      </a>
    </div>
  </div>
</div>
`;
